### ISSUE TITLE
Problems building on Centos7

### ISSUE THREAD

**Issue by dkrovich** [kubernetes/kubernetes#27572]
Did a git clone and trying to build on centos.

[root@localup-cluster kubernetes]# cat /etc/redhat-release 
CentOS Linux release 7.2.1511 (Core) 

[root@localup-cluster kubernetes]# make release
build/release.sh
+++ [0616 17:42:23] Verifying Prerequisites....
+++ [0616 17:42:57] Building Docker image kube-build:build-5a024dbeeb.
+++ [0616 17:44:44] Running build command....
Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
Go version: go version go1.6.2 linux/amd64
+++ [0616 17:45:01] Multiple platforms requested, but available 0G < threshold 11G, building platforms in serial
+++ [0616 17:45:01] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [0616 17:45:04] Building go targets for linux/amd64:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubemark
    cmd/hyperkube
    federation/cmd/federation-apiserver
    federation/cmd/federation-controller-manager
    plugin/cmd/kube-scheduler
go build k8s.io/kubernetes/vendor/github.com/vmware/govmomi/vim25/types: /usr/local/go/pkg/tool/linux_amd64/compile: signal: killed
!!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:491
  'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${goldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:491 kube::golang::build_binaries_for_platform(...)
  2: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:660 kube::golang::build_binaries(...)
  3: hack/build-cross.sh:28 main(...)
Exiting with status 1
!!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:577
  '( kube::golang::setup_env; echo "Go version: $(go version)"; local host_platform; host_platform=$(kube::golang::host_platform); local goflags goldflags; eval "goflags=(${KUBE_GOFLAGS:-})"; goldflags="${KUBE_GOLDFLAGS:-} $(kube::version::ldflags)"; local use_go_build; local -a targets=(); local arg; for arg in "$@";
do
    if [[ "${arg}" == "--use_go_build" ]]; then
        use_go_build=true;
    else
        if [[ "${arg}" == -\* ]]; then
            goflags+=("${arg}");
        else
            targets+=("${arg}");
        fi;
    fi;
done; if [[ ${#targets[@]} -eq 0 ]]; then
    targets=("${KUBE_ALL_TARGETS[@]}");
fi; local -a platforms=("${KUBE_BUILD_PLATFORMS[@]:+${KUBE_BUILD_PLATFORMS[@]}}"); if [[ ${#platforms[@]} -eq 0 ]]; then
    platforms=("${host_platform}");
fi; local binaries; binaries=($(kube::golang::binaries_from_targets "${targets[@]}")); local parallel=false; if [[ ${#platforms[@]} -gt 1 ]]; then
    local gigs; gigs=$(kube::golang::get_physmem); if [[ ${gigs} -ge ${KUBE_PARALLEL_BUILD_MEMORY} ]]; then
        kube::log::status "Multiple platforms requested and available ${gigs}G >= threshold ${KUBE_PARALLEL_BUILD_MEMORY}G, building platforms in parallel"; parallel=true;
    else
        kube::log::status "Multiple platforms requested, but available ${gigs}G < threshold ${KUBE_PARALLEL_BUILD_MEMORY}G, building platforms in serial"; parallel=false;
    fi;
fi; kube::golang::build_kube_toolchain; if [[ "${parallel}" == "true" ]]; then
    kube::log::status "Building go targets for ${platforms[@]} in parallel (output will appear in a burst when complete):" "${targets[@]}"; local platform; for platform in "${platforms[@]}";
    do
        ( kube::golang::set_platform_envs "${platform}"; kube::log::status "${platform}: go build started"; kube::golang::build_binaries_for_platform ${platform} ${use_go_build:-}; kube::log::status "${platform}: go build finished" ) &> "/tmp//${platform//\//_}.build" &
    done; local fails=0; for job in $(jobs -p);
    do
        wait ${job} || let "fails+=1";
    done; for platform in "${platforms[@]}";
    do
        cat "/tmp//${platform//\//_}.build";
    done; exit ${fails};
else
    for platform in "${platforms[@]}";
    do
        kube::log::status "Building go targets for ${platform}:" "${targets[@]}"; kube::golang::set_platform_envs "${platform}"; kube::golang::build_binaries_for_platform ${platform} ${use_go_build:-};
    done;
fi )' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:577 kube::golang::build_binaries(...)
  2: hack/build-cross.sh:28 main(...)
Exiting with status 1
!!! Error in build/../build/common.sh:616
  '"${docker_cmd[@]}" "$@"' exited with status 1
Call stack:
  1: build/../build/common.sh:616 kube::build::run_build_command(...)
  2: build/release.sh:33 main(...)
Exiting with status 1
make: **\* [release] Error 1

**Comment by metal3d** [comment]
Note that Centos download a bad "docker" binary that is an image with "404 Error".
Centos seems abandonned by Kubernetes team... so I needed to make the entire operation with Ansible. And that is not simple at all :)

**Comment by xiangpengzhao** [comment]
There was below info from your log:
`+++ [0616 17:45:01] Multiple platforms requested, but available 0G < threshold 11G, building platforms in serial`

```
go build k8s.io/kubernetes/vendor/github.com/vmware/govmomi/vim25/types: 
/usr/local/go/pkg/tool/linux_amd64/compile: 
signal: killed
```

It seems like that the compiler was OOM killed due to memory pressure. AFAIK, `make release` consumes a lot of memory. I used to encounter `cannot allocate memory` when running it.

We're going to close this issue. Feel free to file a new issue if you fail to run `make release` at latest HEAD and you think there is a bug. Thanks!

@thockin
